app/src/lib/git/status.ts

```diff
@@ -109,83 +116,36 @@ export async function getStatus(
-    repository.path
+  // run git diff check if anything is conflicted
+  const filesWithConflictMarkers = entries.some(
+    es => mapStatus(es.statusCode).kind === 'conflicted'
   )
+    ? await getFilesWithConflictMarkers(repository.path)
+    : new Set<string>()
 
   // Map of files keyed on their paths.
   // Note, map maintains insertion order
-  const files = new Map<string, WorkingDirectoryFileChange>()
-
-  let currentBranch: string | undefined = undefined
-  let currentUpstreamBranch: string | undefined = undefined
-  let currentTip: string | undefined = undefined
-  let branchAheadBehind: IAheadBehind | undefined = undefined
-
-  for (const entry of parsePorcelainStatus(stdout)) {
-    if (entry.kind === 'entry') {
-      const status = mapStatus(entry.statusCode)
-      const hasConflictMarkers = filesWithConflictMarkers.has(entry.path)
-
-      if (status.kind === 'ordinary') {
-        // when a file is added in the index but then removed in the working
-        // directory, the file won't be part of the commit, so we can skip
-        // displaying this entry in the changes list
-        if (
-          status.index === GitStatusEntry.Added &&
-          status.workingTree === GitStatusEntry.Deleted
-        ) {
-          continue
-        }
-      }
-
-      if (status.kind === 'untracked') {
-        // when a delete has been staged, but an untracked file exists with the
-        // same path, we should ensure that we only draw one entry in the
-        // changes list - see if an entry already exists for this path and
-        // remove it if found
-        files.delete(entry.path)
-      }
-
-      // for now we just poke at the existing summary
-      const summary = convertToAppStatus(status, hasConflictMarkers)
-      const selection = DiffSelection.fromInitialSelection(
-        DiffSelectionType.All
-      )
-
-      files.set(
-        entry.path,
-        new WorkingDirectoryFileChange(
-          entry.path,
-          summary,
-          selection,
-          entry.oldPath
-        )
-      )
-    } else if (entry.kind === 'header') {
-      let m: RegExpMatchArray | null
-      const value = entry.value
-
-      // This intentionally does not match branch.oid initial
-      if ((m = value.match(/^branch\.oid ([a-f0-9]+)$/))) {
-        currentTip = m[1]
-      } else if ((m = value.match(/^branch.head (.*)/))) {
-        if (m[1] !== '(detached)') {
-          currentBranch = m[1]
-        }
-      } else if ((m = value.match(/^branch.upstream (.*)/))) {
-        currentUpstreamBranch = m[1]
-      } else if ((m = value.match(/^branch.ab \+(\d+) -(\d+)$/))) {
-        const ahead = parseInt(m[1], 10)
-        const behind = parseInt(m[2], 10)
-
-        if (!isNaN(ahead) && !isNaN(behind)) {
-          branchAheadBehind = { ahead, behind }
-        }
-      }
-    }
-  }
+  const files = entries.reduce(
+    (files, entry) => addEntryToFiles(files, entry, filesWithConflictMarkers),
```

Comment:
:art: defining `files` as variables in a couple of places is probably fine from a compiler side (it usually gets noisy about shadowed variables when it hits them), but from a readability side could we change the inner `files` to something else to indicate the scope here is different?